### PR TITLE
Autodetect unused disks for SR tests (sr_disk fixture and such)

### PR DESCRIPTION
**Pull request by stormi**
And more or less related changes.

Depends on #73

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -184,15 +184,22 @@ def sr_disk_wiped(host, sr_disk):
     yield sr_disk
 
 @pytest.fixture(scope='session')
-def sr_disk_for_all_hosts(host, sr_disk):
+def sr_disk_for_all_hosts(host):
+    master_disks = host.available_disks()
+    assert len(master_disks) > 0, "a free disk device is required on the master host"
```
Comment (benjamreis) [inline conftest.py:189]: the assert can be put in `available_disks` perhaps with a `raise` flag. To avoir duplicating the assert and error message.

**Review by benjamreis — CHANGES_REQUESTED** [review]
A lot of modifs come from the split of `common.py`, please re ask for a review once, this PR is rebased after the split has been merged.

File: conftest.py (hunk 1 above)
Comment (Wescoeur) [reply conftest.py:189]: I'm not sure it's a good idea: `available_disks` is clear, it returns the available disks, IMHO an assert in this helper is equal to a bad naming.

Comment (Wescoeur) [reply conftest.py:189]: Also `available_disks` is used in the for loop after this line without `assert`. ;)

Comment (stormi) [reply conftest.py:189]: If there's no consensus, then I guess I won't change it :D

Comment (benjamreis) [reply conftest.py:189]: Tht's why I proposed a flag to param it ;)

Comment (Wescoeur) [reply conftest.py:189]: Sometimes, it's not a problem to duplicate a line, if it's not abusive. ^^"

**Comment by stormi** [comment]
I ran storage tests (except CephFS, Linstor and MooseFS for whose we need make it easier to run the tests internally) with success (with a minimal VM).

**Review by benjamreis — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

**Pull request merged**